crates/worker/src/partition/leadership/leader_state.rs

```diff
@@ -143,7 +143,17 @@ impl LeaderState {
         );
         let mut all_streams = all_streams.ready_chunks(BATCH_READY_UP_TO);
 
+        let shuffle_task_handle = self.shuffle_task_handle.as_mut().expect("is set");
         tokio::select! {
+            // watch the shuffle task in case it crashed
+            result = shuffle_task_handle => {
+                self.shuffle_task_handle.take();
```

Comment:
Are we taking the shuffle task here so that we don't poll it again on `stop`? If yes, then maybe add a comment to the field in the struct explaining this.

Reply:
Yes, it's illegal to pull a handle once it has been resolved. I will add the comment